molr-mole-core/src/main/java/io/molr/mole/core/tree/TreeMissionExecutor.java

```diff
@@ -94,6 +103,13 @@ private MissionState gatherMissionState() {
 
         resultTracker.blockResults().entrySet().forEach(e -> builder.blockResult(e.getKey(), e.getValue()));
         runStateTracker.blockResults().entrySet().forEach(e -> builder.blockRunState(e.getKey(), e.getValue()));
+        
+        breakpoints.forEach(builder::addBreakpoint);
+        representation.allBlocks().forEach(block -> {
+            builder.addAllowedCommand(block, BlockCommand.UNSET_BREAKPOINT);
```

Comment:
Consider being more sensitive here: Only blocks with no breakpoint should allow adding one, blocks with breakpoints should allow removing it.